include/circt/Dialect/SV/SVInOutOps.td

```diff
@@ -137,3 +139,66 @@ def ArrayIndexInOutOp
   let assemblyFormat =
     "$input`[`$index`]` attr-dict `:` type($input) `,` type($index)";
 }
+
+def IndexedPartSelectInOutOp : SVOp<"indexed_part_select", [NoSideEffect, InferTypeOpInterface]> {
+  let summary = "Address several contiguous bits of an inout type (e.g. a wire"
+                " or inout port). This is an indexed part-select operator."
+                "The base is an integer expression and the width is an "
+                " integer constant. The bits start from base and the number "
+                "of bits selected is equal to width. If $decrement is true, "
+                " then part select decrements starting from $base."
+                "See SV Spec 11.5.1.";
+
+  let arguments = (ins InOutType:$input, HWIntegerType:$base, I32Attr:$width,
+                      UnitAttr:$decrement);
+  let results = (outs InOutType:$result);
+
+  let builders = [
+    OpBuilder<(ins "Value":$input, "Value":$base, "int32_t":$width)>,
+    OpBuilder<(ins "Value":$input, "Value":$base, "int32_t":$width,
+                                                "bool":$decrement)>
+  ];
+
+  let extraClassDeclaration = [{
+    /// Infer the return types of this operation.
+    static LogicalResult inferReturnTypes(MLIRContext *context,
+                                          Optional<Location> loc,
+                                          ValueRange operands,
+                                          DictionaryAttr attrs,
+                                          mlir::RegionRange regions,
+                                          SmallVectorImpl<Type> &results);
+  }];
+
+  let assemblyFormat = "$input`[`$base (`decrement` $decrement^)?`:` $width`]`"
+                        " attr-dict `:` type($input) `,` type($base)";
+}
+
+
+def IndexedPartSelectReadInOutOp
```

Comment:
There doesn't need to be a read op, there is already a read op for inout values.   There should be a completely non-inout version of the op.